{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# iHART\n",
    "Get balanced samples for sex-associated coverage.\n",
    "- 1 neurotypical male + 1 neurotypical female from each family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-2-1dffac2cc967>, line 3)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-2-1dffac2cc967>\"\u001b[0;36m, line \u001b[0;32m3\u001b[0m\n\u001b[0;31m    aws s3 ls ihart-hg38/cram/ |  awk '{ print $4 }'  > /scratch/groups/dpwall/personal/briannac/unmapped_reads/data/file_dirs_on_aws.tsv\u001b[0m\n\u001b[0m         ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "# Get files actually on AWS \n",
    "%%bash\n",
    "aws s3 ls ihart-hg38/cram/ |  awk '{ print $4 }'  > /scratch/groups/dpwall/personal/briannac/unmapped_reads/data/file_dirs_on_aws.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "bam_mappings = pd.read_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/bam_mappings.csv', sep='\\t')\n",
    "bam_mappings.index = bam_mappings.sample_id\n",
    "aws_files= pd.read_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/file_dirs_on_aws.tsv', header=None)[0].values\n",
    "bam_mappings = bam_mappings.loc[list(set(bam_mappings.index).intersection(set([a.replace('.final.cram', '').replace('.crai', '') for a in aws_files])))]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "children = bam_mappings[(bam_mappings.relationship=='sibling') & (pd.isna(bam_mappings.derived_affected_status))]\n",
    "parents = bam_mappings[(bam_mappings.relationship!='sibling') & (pd.isna(bam_mappings.derived_affected_status))]\n",
    "\n",
    "both_sex_fams = children.groupby('family').aggregate(lambda x: ('1.0' in list(x)) & ('2.0' in list(x)))\n",
    "both_parents_fams = parents.groupby('family').aggregate(lambda x: ('mother' in list(x)) & ('father' in list(x)))\n",
    "\n",
    "children = children[[((f in set(both_sex_fams[both_sex_fams.sex_numeric].index))\n",
    "                     & (f in set(both_parents_fams[both_parents_fams.relationship].index)))  for f in children.family.values]]\n",
    "males = children[children.sex_numeric=='1.0']\n",
    "_, idx = np.unique(males.family, return_index=True)\n",
    "unaff_males = males.iloc[idx]\n",
    "females = children[children.sex_numeric=='2.0']\n",
    "_, idx = np.unique(females.family, return_index=True)\n",
    "unaff_females = females.iloc[idx]\n",
    "\n",
    "\n",
    "males = children[children.sex_numeric=='1.0']\n",
    "_, idx = np.unique(males.family, return_index=True)\n",
    "aff_males = males.iloc[idx]\n",
    "females = children[children.sex_numeric=='2.0']\n",
    "_, idx = np.unique(females.family, return_index=True)\n",
    "aff_females = females.iloc[idx]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "children = bam_mappings[(bam_mappings.relationship=='sibling') & (pd.isna(bam_mappings.derived_affected_status))]\n",
    "parents = bam_mappings[(bam_mappings.relationship!='sibling')]\n",
    "children = bam_mappings[(bam_mappings.relationship=='sibling') & (pd.isna(bam_mappings.derived_affected_status))]\n",
    "\n",
    "both_sex_fams = children.groupby('family').aggregate(lambda x: ('1.0' in list(x)) & ('2.0' in list(x)))\n",
    "same_val_fams = children.groupby('family').aggregate(lambda x: len(set(x))==1)\n",
    "both_parents_fams = parents.groupby('family').aggregate(lambda x: ('mother' in list(x)) & ('father' in list(x)) & (len(x)==2))\n",
    "\n",
    "children = children[[(\n",
    "    (f in set(both_sex_fams[both_sex_fams.sex_numeric].index)) & \n",
    "    (f in set(same_val_fams[same_val_fams.race].index)) & \n",
    "    (f in set(same_val_fams[same_val_fams.race_ethnic].index)) & \n",
    "    (f in set(both_parents_fams[both_parents_fams.relationship].index)) &\n",
    "    (f in set(same_val_fams[same_val_fams.FatherID].index)) &\n",
    "    (f in set(same_val_fams[same_val_fams.MotherID].index))) for f in children.family.values]]\n",
    "\n",
    "males = children[children.sex_numeric=='1.0']\n",
    "_, idx = np.unique(males.family, return_index=True)\n",
    "unaff_males = males.iloc[idx].sort_values(['race', 'race_ethnic', 'family'])\n",
    "females = children[children.sex_numeric=='2.0']\n",
    "_, idx = np.unique(females.family, return_index=True)\n",
    "unaff_females = females.iloc[idx].sort_values(['race', 'race_ethnic', 'family'])\n",
    "\n",
    "\n",
    "children = bam_mappings[(bam_mappings.relationship=='sibling') & (~pd.isna(bam_mappings.derived_affected_status))]\n",
    "\n",
    "both_sex_fams = children.groupby('family').aggregate(lambda x: ('1.0' in list(x)) & ('2.0' in list(x)))\n",
    "same_val_fams = children.groupby('family').aggregate(lambda x: len(set(x))==1)\n",
    "both_parents_fams = parents.groupby('family').aggregate(lambda x: ('mother' in list(x)) & ('father' in list(x)) & (len(x)==2))\n",
    "\n",
    "children = children[[(\n",
    "    (f in set(both_sex_fams[both_sex_fams.sex_numeric].index)) & \n",
    "    (f in set(same_val_fams[same_val_fams.race].index)) & \n",
    "    (f in set(same_val_fams[same_val_fams.derived_affected_status].index)) & \n",
    "    (f in set(same_val_fams[same_val_fams.race_ethnic].index)) & \n",
    "    (f in set(both_parents_fams[both_parents_fams.relationship].index)) &\n",
    "    (f in set(same_val_fams[same_val_fams.FatherID].index)) &\n",
    "    (f in set(same_val_fams[same_val_fams.MotherID].index))) for f in children.family.values]]\n",
    "\n",
    "males = children[children.sex_numeric=='1.0']\n",
    "_, idx = np.unique(males.family, return_index=True)\n",
    "aff_males = males.iloc[idx].sort_values(['race', 'race_ethnic', 'family'])\n",
    "females = children[children.sex_numeric=='2.0']\n",
    "_, idx = np.unique(females.family, return_index=True)\n",
    "aff_females = females.iloc[idx].sort_values(['race', 'race_ethnic', 'family'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "212"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(aff_females)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "selected_children = pd.concat([unaff_males, unaff_females, aff_males.iloc[:100], aff_females[:100]])\n",
    "selected_parents = parents[[f in set(selected_children.family) for f in parents.family]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALL crams copied!\n"
     ]
    }
   ],
   "source": [
    "## Check if all CRAMS are copied.\n",
    "\n",
    "for iteration, i in enumerate(selected_children.index):\n",
    "    if not len(glob.glob('/scratch/groups/dpwall/personal/briannac/unmapped_reads/pileups/data/crams/' + i + '.cram')):\n",
    "        print(i, iteration+1)\n",
    "        \n",
    "for iteration, i in enumerate(selected_parents.index):\n",
    "    if not len(glob.glob('/scratch/groups/dpwall/personal/briannac/unmapped_reads/pileups/data/crams/' + i + '.cram')):\n",
    "        print(i, iteration+1)\n",
    "print(\"ALL crams copied!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/users/briannac/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "selected_children['file_loc'] = ['s3://ihart-hg38/cram/%s.final.cram' %i for i in selected_children.index]\n",
    "selected_parents['file_loc'] = ['s3://ihart-hg38/cram/%s.final.cram' %i for i in selected_parents.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/users/briannac/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "selected_children['batch'] = ['batch_' + b.split('-')[-1] for b in selected_children.sequencing_plate.values]\n",
    "selected_parents['batch'] = ['batch_' + b.split('-')[-1] for b in selected_parents.sequencing_plate.values]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "selected_children[['file_loc', 'batch']].to_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/balanced_children_ihart.csv',\n",
    "                                   header=None, sep='\\t')\n",
    "selected_parents[['file_loc', 'batch']].to_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/balanced_parents_ihart.csv',\n",
    "                                   header=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "### parents + children dataframe (sorted)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "children_df = pd.read_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/balanced_children_ihart.csv', sep='\\t', header=None)\n",
    "parents_df = pd.read_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/balanced_parents_ihart.csv', sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "children_and_parents_df = pd.concat([children_df, parents_df])\n",
    "children_and_parents_df.index = children_and_parents_df[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "children_and_parents_df = children_and_parents_df.loc[\n",
    "    list(bam_mappings.loc[children_and_parents_df[0]].sort_values('family').index)].drop([0], axis=1)\n",
    "children_and_parents_df.to_csv(\n",
    "    '/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/parents_and_children.csv', header=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 10000 Genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Print ids for copy paste in bash script.\n",
    "idx = [g.split('/')[-1].split('.final')[0] for g in glob.glob(\n",
    "    '/scratch/groups/dpwall/personal/chloehe/unmapped_reads/bam/batch_00514/*.final.cram')]\n",
    "' '.join(idx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "kg_ped = pd.read_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/1kg_related_samples.ped', sep=' ')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(('0', '0'), 2590),\n",
       " (('HG00656', 'HG00657'), 2),\n",
       " (('HG03943', 'HG03944'), 2),\n",
       " (('HG03679', 'HG03642'), 2),\n",
       " (('NA19661', 'NA19660'), 2),\n",
       " (('NA19679', 'NA19678'), 2),\n",
       " (('0', 'HG00144'), 1),\n",
       " (('HG00403', 'HG00404'), 1),\n",
       " (('HG00406', 'HG00407'), 1),\n",
       " (('HG00418', 'HG00419'), 1),\n",
       " (('HG00421', 'HG00422'), 1),\n",
       " (('HG00427', 'HG00428'), 1),\n",
       " (('HG00436', 'HG00437'), 1),\n",
       " (('HG00442', 'HG00443'), 1),\n",
       " (('HG00445', 'HG00446'), 1),\n",
       " (('HG00448', 'HG00449'), 1),\n",
       " (('HG00451', 'HG00452'), 1),\n",
       " (('HG00457', 'HG00458'), 1),\n",
       " (('HG00463', 'HG00464'), 1),\n",
       " (('HG00472', 'HG00473'), 1),\n",
       " (('HG00475', 'HG00476'), 1),\n",
       " (('HG00478', 'HG00479'), 1),\n",
       " (('HG00500', 'HG00501'), 1),\n",
       " (('HG00512', 'HG00513'), 1),\n",
       " (('HG00524', 'HG00525'), 1),\n",
       " (('HG00530', 'HG00531'), 1),\n",
       " (('HG00533', 'HG00534'), 1),\n",
       " (('HG00536', 'HG00537'), 1),\n",
       " (('HG00542', 'HG00543'), 1),\n",
       " (('HG01075', 'HG00551'), 1),\n",
       " (('HG00553', 'HG00554'), 1),\n",
       " (('HG00556', 'HG00557'), 1),\n",
       " (('HG00559', 'HG00560'), 1),\n",
       " (('HG00565', 'HG00566'), 1),\n",
       " (('HG00577', 'HG00578'), 1),\n",
       " (('HG00580', 'HG00581'), 1),\n",
       " (('HG00583', 'HG00584'), 1),\n",
       " (('HG00589', 'HG00590'), 1),\n",
       " (('HG00592', 'HG00593'), 1),\n",
       " (('HG00595', 'HG00596'), 1),\n",
       " (('HG00607', 'HG00608'), 1),\n",
       " (('HG00610', 'HG00611'), 1),\n",
       " (('HG00613', 'HG00614'), 1),\n",
       " (('HG00619', 'HG00620'), 1),\n",
       " (('HG00625', 'HG00626'), 1),\n",
       " (('HG00628', 'HG00629'), 1),\n",
       " (('HG00634', 'HG00635'), 1),\n",
       " (('HG00637', 'HG00638'), 1),\n",
       " (('HG00640', 'HG00641'), 1),\n",
       " (('HG00650', 'HG00651'), 1),\n",
       " (('HG00653', 'HG00654'), 1),\n",
       " (('HG00662', 'HG00663'), 1),\n",
       " (('HG00671', 'HG00672'), 1),\n",
       " (('HG00683', 'HG00684'), 1),\n",
       " (('HG00689', 'HG00690'), 1),\n",
       " (('HG00692', 'HG00693'), 1),\n",
       " (('HG00698', 'HG00699'), 1),\n",
       " (('HG00701', 'HG00702'), 1),\n",
       " (('HG00704', 'HG00705'), 1),\n",
       " (('HG00707', 'HG00708'), 1),\n",
       " (('HG00731', 'HG00732'), 1),\n",
       " (('HG01047', 'HG00734'), 1),\n",
       " (('HG00736', 'HG00737'), 1),\n",
       " (('HG00739', 'HG00740'), 1),\n",
       " (('HG00866', '0'), 1),\n",
       " (('HG01048', 'HG01049'), 1),\n",
       " (('HG01051', 'HG01052'), 1),\n",
       " (('HG01054', 'HG01055'), 1),\n",
       " (('HG01060', 'HG01061'), 1),\n",
       " (('HG01066', 'HG01067'), 1),\n",
       " (('HG01069', 'HG01070'), 1),\n",
       " (('HG01072', 'HG01073'), 1),\n",
       " (('HG01079', 'HG01080'), 1),\n",
       " (('HG01082', 'HG01083'), 1),\n",
       " (('HG01085', 'HG01086'), 1),\n",
       " (('HG01094', 'HG01095'), 1),\n",
       " (('HG01097', 'HG01098'), 1),\n",
       " (('HG01247', 'HG01248'), 1),\n",
       " (('HG01101', 'HG01102'), 1),\n",
       " (('HG01104', 'HG01105'), 1),\n",
       " (('HG01107', 'HG01108'), 1),\n",
       " (('HG01112', 'HG01113'), 1),\n",
       " (('HG01124', 'HG01125'), 1),\n",
       " (('HG01133', 'HG01134'), 1),\n",
       " (('HG01136', 'HG01137'), 1),\n",
       " (('HG01139', 'HG01140'), 1),\n",
       " (('HG01148', 'HG01149'), 1),\n",
       " (('HG01167', 'HG01168'), 1),\n",
       " (('HG01170', 'HG01171'), 1),\n",
       " (('HG01173', 'HG01174'), 1),\n",
       " (('HG01176', 'HG01177'), 1),\n",
       " (('HG01182', 'HG01183'), 1),\n",
       " (('HG01187', 'HG01188'), 1),\n",
       " (('HG01190', 'HG01191'), 1),\n",
       " (('HG01197', 'HG01198'), 1),\n",
       " (('HG01204', 'HG01205'), 1),\n",
       " (('HG01241', 'HG01242'), 1),\n",
       " (('HG01110', 'HG01111'), 1),\n",
       " (('HG01250', 'HG01251'), 1),\n",
       " (('HG01253', 'HG01254'), 1),\n",
       " (('HG01256', 'HG01257'), 1),\n",
       " (('HG01259', 'HG01260'), 1),\n",
       " (('HG01271', 'HG01272'), 1),\n",
       " (('HG01274', 'HG01275'), 1),\n",
       " (('HG01277', 'HG01278'), 1),\n",
       " (('HG01341', 'HG01342'), 1),\n",
       " (('HG01344', 'HG01345'), 1),\n",
       " (('HG01347', 'HG01348'), 1),\n",
       " (('HG01350', 'HG01351'), 1),\n",
       " (('HG01353', 'HG01354'), 1),\n",
       " (('HG01356', 'HG01357'), 1),\n",
       " (('HG01359', 'HG01360'), 1),\n",
       " (('HG01365', 'HG01366'), 1),\n",
       " (('HG01374', 'HG01375'), 1),\n",
       " (('HG01377', 'HG01378'), 1),\n",
       " (('HG01383', 'HG01384'), 1),\n",
       " (('HG01389', 'HG01390'), 1),\n",
       " (('HG01431', 'HG01432'), 1),\n",
       " (('HG01437', 'HG01438'), 1),\n",
       " (('HG01455', 'HG01456'), 1),\n",
       " (('HG01461', 'HG01462'), 1),\n",
       " (('HG01464', 'HG01465'), 1),\n",
       " (('HG01488', 'HG01489'), 1),\n",
       " (('HG01491', 'HG01492'), 1),\n",
       " (('HG01494', 'HG01495'), 1),\n",
       " (('HG01497', 'HG01498'), 1),\n",
       " (('HG01500', 'HG01501'), 1),\n",
       " (('HG01503', 'HG01504'), 1),\n",
       " (('HG01506', 'HG01507'), 1),\n",
       " (('HG01509', 'HG01510'), 1),\n",
       " (('HG01512', 'HG01513'), 1),\n",
       " (('HG01515', 'HG01516'), 1),\n",
       " (('HG01518', 'HG01519'), 1),\n",
       " (('HG01521', 'HG01522'), 1),\n",
       " (('HG01524', 'HG01525'), 1),\n",
       " (('HG01527', 'HG01528'), 1),\n",
       " (('HG01530', 'HG01531'), 1),\n",
       " (('HG01536', 'HG01537'), 1),\n",
       " (('HG01550', 'HG01551'), 1),\n",
       " (('HG01565', 'HG01566'), 1),\n",
       " (('HG01571', 'HG01572'), 1),\n",
       " (('HG01577', 'HG01578'), 1),\n",
       " (('HG01603', 'HG01602'), 1),\n",
       " (('HG01606', 'HG01605'), 1),\n",
       " (('HG01608', 'HG01607'), 1),\n",
       " (('HG01610', 'HG01612'), 1),\n",
       " (('HG01615', 'HG01613'), 1),\n",
       " (('HG01617', 'HG01618'), 1),\n",
       " (('HG01619', 'HG01620'), 1),\n",
       " (('HG01624', 'HG01623'), 1),\n",
       " (('HG01625', 'HG01626'), 1),\n",
       " (('HG01630', 'HG01628'), 1),\n",
       " (('HG01631', 'HG01632'), 1),\n",
       " (('HG01669', 'HG01668'), 1),\n",
       " (('HG01672', 'HG01670'), 1),\n",
       " (('HG01675', 'HG01673'), 1),\n",
       " (('HG01678', 'HG01676'), 1),\n",
       " (('HG01680', 'HG01679'), 1),\n",
       " (('HG01682', 'HG01684'), 1),\n",
       " (('HG01686', 'HG01685'), 1),\n",
       " (('HG01694', 'HG01695'), 1),\n",
       " (('HG01699', 'HG01697'), 1),\n",
       " (('HG01700', 'HG01702'), 1),\n",
       " (('HG01705', 'HG01704'), 1),\n",
       " (('HG01708', 'HG01707'), 1),\n",
       " (('HG01709', 'HG01710'), 1),\n",
       " (('HG01747', 'HG01746'), 1),\n",
       " (('HG01756', 'HG01757'), 1),\n",
       " (('HG01761', 'HG01762'), 1),\n",
       " (('HG01771', 'HG01770'), 1),\n",
       " (('HG01775', 'HG01773'), 1),\n",
       " (('HG01777', 'HG01776'), 1),\n",
       " (('HG01781', 'HG01779'), 1),\n",
       " (('HG01783', 'HG01784'), 1),\n",
       " (('HG01879', 'HG01880'), 1),\n",
       " (('HG01885', 'HG01956'), 1),\n",
       " (('HG02014', 'HG01886'), 1),\n",
       " (('HG01882', 'HG01883'), 1),\n",
       " (('HG01890', 'HG01889'), 1),\n",
       " (('HG01912', 'HG01894'), 1),\n",
       " (('HG02013', 'HG01896'), 1),\n",
       " (('HG01892', 'HG01893'), 1),\n",
       " (('HG01914', 'HG01915'), 1),\n",
       " (('HG01917', 'HG01918'), 1),\n",
       " (('HG01920', 'HG01921'), 1),\n",
       " (('HG01923', 'HG01924'), 1),\n",
       " (('HG01926', 'HG01927'), 1),\n",
       " (('HG01932', 'HG01933'), 1),\n",
       " (('HG01935', 'HG01936'), 1),\n",
       " (('HG01938', 'HG01939'), 1),\n",
       " (('HG01941', 'HG01942'), 1),\n",
       " (('HG01944', 'HG01945'), 1),\n",
       " (('HG01947', 'HG01948'), 1),\n",
       " (('HG01950', 'HG01951'), 1),\n",
       " (('HG01953', 'HG01954'), 1),\n",
       " (('HG01986', 'HG01958'), 1),\n",
       " (('HG01988', 'HG01989'), 1),\n",
       " (('HG01967', 'HG01968'), 1),\n",
       " (('HG01970', 'HG01971'), 1),\n",
       " (('HG01974', 'HG01973'), 1),\n",
       " (('HG01977', 'HG01976'), 1),\n",
       " (('HG01979', 'HG01980'), 1),\n",
       " (('HG01982', 'HG01983'), 1),\n",
       " (('HG01990', 'HG02012'), 1),\n",
       " (('HG01991', 'HG01992'), 1),\n",
       " (('HG02008', 'HG01997'), 1),\n",
       " (('HG02002', 'HG02003'), 1),\n",
       " (('HG02009', 'HG02010'), 1),\n",
       " (('HG02017', 'HG02016'), 1),\n",
       " (('HG02020', 'HG02019'), 1),\n",
       " (('HG02026', 'HG02025'), 1),\n",
       " (('HG02029', 'HG02028'), 1),\n",
       " (('HG02032', 'HG02031'), 1),\n",
       " (('HG02053', 'HG02054'), 1),\n",
       " (('HG02058', 'HG02057'), 1),\n",
       " (('HG02061', 'HG02060'), 1),\n",
       " (('HG02070', 'HG02069'), 1),\n",
       " (('HG02073', 'HG02072'), 1),\n",
       " (('HG02076', 'HG02075'), 1),\n",
       " (('HG02079', 'HG02078'), 1),\n",
       " (('HG02082', 'HG02081'), 1),\n",
       " (('HG02085', 'HG02084'), 1),\n",
       " (('HG02090', 'HG02089'), 1),\n",
       " (('HG02104', 'HG02105'), 1),\n",
       " (('HG02122', 'HG02121'), 1),\n",
       " (('HG02128', 'HG02127'), 1),\n",
       " (('HG02131', 'HG02130'), 1),\n",
       " (('HG02134', 'HG02133'), 1),\n",
       " (('HG02137', 'HG02136'), 1),\n",
       " (('HG02143', 'HG02144'), 1),\n",
       " (('HG02146', 'HG02147'), 1),\n",
       " (('HG02221', 'HG02220'), 1),\n",
       " (('HG02224', 'HG02223'), 1),\n",
       " (('HG02231', 'HG02230'), 1),\n",
       " (('HG02233', 'HG02232'), 1),\n",
       " (('HG02236', 'HG02235'), 1),\n",
       " (('HG02238', 'HG02239'), 1),\n",
       " (('HG02255', 'HG02256'), 1),\n",
       " (('HG02314', 'HG02315'), 1),\n",
       " (('HG02259', 'HG02260'), 1),\n",
       " (('HG02271', 'HG02272'), 1),\n",
       " (('HG02277', 'HG02278'), 1),\n",
       " (('HG02307', 'HG02308'), 1),\n",
       " (('HG02285', 'HG02286'), 1),\n",
       " (('HG02291', 'HG02292'), 1),\n",
       " (('HG02299', 'HG02298'), 1),\n",
       " (('HG02302', 'HG02301'), 1),\n",
       " (('HG02317', 'HG02318'), 1),\n",
       " (('HG02281', 'HG02282'), 1),\n",
       " (('HG02449', 'HG02450'), 1),\n",
       " (('HG02461', 'HG02462'), 1),\n",
       " (('HG02464', 'HG02465'), 1),\n",
       " (('HG02490', 'HG02491'), 1),\n",
       " (('HG02493', 'HG02494'), 1),\n",
       " (('HG02512', 'HG02513'), 1),\n",
       " (('HG02521', 'HG02522'), 1),\n",
       " (('HG02524', 'HG02525'), 1),\n",
       " (('HG02561', 'HG02562'), 1),\n",
       " (('HG02567', 'HG02568'), 1),\n",
       " (('HG02570', 'HG02571'), 1),\n",
       " (('HG02573', 'HG02574'), 1),\n",
       " (('HG02582', 'HG02583'), 1),\n",
       " (('HG02585', 'HG02586'), 1),\n",
       " (('HG02588', 'HG02589'), 1),\n",
       " (('HG02594', 'HG02595'), 1),\n",
       " (('HG02600', 'HG02601'), 1),\n",
       " (('HG02603', 'HG02604'), 1),\n",
       " (('HG02610', 'HG02611'), 1),\n",
       " (('HG02613', 'HG02614'), 1),\n",
       " (('HG02620', 'HG02621'), 1),\n",
       " (('HG02623', 'HG02624'), 1),\n",
       " (('HG02628', 'HG02629'), 1),\n",
       " (('HG02634', 'HG02635'), 1),\n",
       " (('HG02642', 'HG02643'), 1),\n",
       " (('HG02645', 'HG02646'), 1),\n",
       " (('HG02648', 'HG02649'), 1),\n",
       " (('HG02651', 'HG02652'), 1),\n",
       " (('HG02654', 'HG02655'), 1),\n",
       " (('HG02657', 'HG02658'), 1),\n",
       " (('HG02660', 'HG02661'), 1),\n",
       " (('HG02666', 'HG02667'), 1),\n",
       " (('HG02675', 'HG02676'), 1),\n",
       " (('HG02678', 'HG02679'), 1),\n",
       " (('HG02681', 'HG02682'), 1),\n",
       " (('HG02684', 'HG02685'), 1),\n",
       " (('HG02687', 'HG02688'), 1),\n",
       " (('HG02690', 'HG02691'), 1),\n",
       " (('HG02696', 'HG02697'), 1),\n",
       " (('HG02699', 'HG02700'), 1),\n",
       " (('HG02702', 'HG02703'), 1),\n",
       " (('HG02715', 'HG02716'), 1),\n",
       " (('HG02721', 'HG02722'), 1),\n",
       " (('HG02724', 'HG02725'), 1),\n",
       " (('HG02727', 'HG02728'), 1),\n",
       " (('HG02733', 'HG02734'), 1),\n",
       " (('HG02736', 'HG02737'), 1),\n",
       " (('HG02759', 'HG02760'), 1),\n",
       " (('HG02762', 'HG02763'), 1),\n",
       " (('HG02768', 'HG02769'), 1),\n",
       " (('HG02771', 'HG02772'), 1),\n",
       " (('HG02774', 'HG02775'), 1),\n",
       " (('HG02783', 'HG02784'), 1),\n",
       " (('HG02786', 'HG02787'), 1),\n",
       " (('HG02789', 'HG02790'), 1),\n",
       " (('HG02792', 'HG02793'), 1),\n",
       " (('HG02798', 'HG02799'), 1),\n",
       " (('HG02804', 'HG02805'), 1),\n",
       " (('HG02807', 'HG02808'), 1),\n",
       " (('HG02810', 'HG02811'), 1),\n",
       " (('HG02813', 'HG02814'), 1),\n",
       " (('HG02816', 'HG02817'), 1),\n",
       " (('HG02819', 'HG02820'), 1),\n",
       " (('HG02836', 'HG02837'), 1),\n",
       " (('HG02839', 'HG02840'), 1),\n",
       " (('HG02851', 'HG02852'), 1),\n",
       " (('HG02854', 'HG02855'), 1),\n",
       " (('HG02860', 'HG02861'), 1),\n",
       " (('HG02869', 'HG02870'), 1),\n",
       " (('HG02878', 'HG02879'), 1),\n",
       " (('HG02881', 'HG02882'), 1),\n",
       " (('HG02884', 'HG02885'), 1),\n",
       " (('HG02887', 'HG02888'), 1),\n",
       " (('HG02890', 'HG02891'), 1),\n",
       " (('HG02895', 'HG02896'), 1),\n",
       " (('HG02923', 'HG02922'), 1),\n",
       " (('HG02944', 'HG02943'), 1),\n",
       " (('HG02947', 'HG02946'), 1),\n",
       " (('HG02953', 'HG02952'), 1),\n",
       " (('HG02965', 'HG02964'), 1),\n",
       " (('HG02971', 'HG02970'), 1),\n",
       " (('HG02973', 'HG02974'), 1),\n",
       " (('HG02977', 'HG02976'), 1),\n",
       " (('HG02981', 'HG02979'), 1),\n",
       " (('HG02982', 'HG02983'), 1),\n",
       " (('HG03006', 'HG03007'), 1),\n",
       " (('HG03015', 'HG03016'), 1),\n",
       " (('HG03024', 'HG03025'), 1),\n",
       " (('HG03027', 'HG03028'), 1),\n",
       " (('HG03033', 'HG03034'), 1),\n",
       " (('HG03039', 'HG03040'), 1),\n",
       " (('HG03045', 'HG03046'), 1),\n",
       " (('HG03048', 'HG03049'), 1),\n",
       " (('HG03054', 'HG03055'), 1),\n",
       " (('HG03063', 'HG03064'), 1),\n",
       " (('HG03096', 'HG03097'), 1),\n",
       " (('HG03100', 'HG03099'), 1),\n",
       " (('HG03109', 'HG03108'), 1),\n",
       " (('HG03112', 'HG03111'), 1),\n",
       " (('HG03115', 'HG03114'), 1),\n",
       " (('HG03118', 'HG03117'), 1),\n",
       " (('HG03120', 'HG03121'), 1),\n",
       " (('HG03124', 'HG03123'), 1),\n",
       " (('HG03127', 'HG03126'), 1),\n",
       " (('HG03130', 'HG03129'), 1),\n",
       " (('HG03133', 'HG03132'), 1),\n",
       " (('HG03136', 'HG03135'), 1),\n",
       " (('HG03160', 'HG03159'), 1),\n",
       " (('HG03163', 'HG03162'), 1),\n",
       " (('HG03169', 'HG03168'), 1),\n",
       " (('HG03190', 'HG03189'), 1),\n",
       " (('HG03196', 'HG03195'), 1),\n",
       " (('HG03199', 'HG03198'), 1),\n",
       " (('HG03228', 'HG03229'), 1),\n",
       " (('HG03234', 'HG03235'), 1),\n",
       " (('HG03237', 'HG03238'), 1),\n",
       " (('HG03240', 'HG03241'), 1),\n",
       " (('HG03246', 'HG03247'), 1),\n",
       " (('HG03249', 'HG03250'), 1),\n",
       " (('HG03258', 'HG03259'), 1),\n",
       " (('HG03268', 'HG03267'), 1),\n",
       " (('HG03271', 'HG03270'), 1),\n",
       " (('HG03295', 'HG03294'), 1),\n",
       " (('HG03298', 'HG03297'), 1),\n",
       " (('HG03301', 'HG03300'), 1),\n",
       " (('HG03304', 'HG03303'), 1),\n",
       " (('HG03307', 'HG03306'), 1),\n",
       " (('HG03311', 'HG03309'), 1),\n",
       " (('HG03313', 'HG03312'), 1),\n",
       " (('HG03343', 'HG03342'), 1),\n",
       " (('HG03352', 'HG03351'), 1),\n",
       " (('HG03367', 'HG03366'), 1),\n",
       " (('HG03370', 'HG03369'), 1),\n",
       " (('HG03373', 'HG03372'), 1),\n",
       " (('HG03469', 'HG03470'), 1),\n",
       " (('HG03466', '0'), 1),\n",
       " (('HG03451', 'HG03452'), 1),\n",
       " (('HG03454', 'HG03455'), 1),\n",
       " (('HG03478', 'HG03479'), 1),\n",
       " (('HG03484', 'HG03485'), 1),\n",
       " (('HG03487', 'HG03488'), 1),\n",
       " (('HG03490', 'HG03491'), 1),\n",
       " (('HG03515', 'HG03514'), 1),\n",
       " (('HG03518', 'HG03517'), 1),\n",
       " (('HG03521', 'HG03520'), 1),\n",
       " (('HG03538', 'HG03539'), 1),\n",
       " (('HG03574', 'HG03575'), 1),\n",
       " (('HG03577', 'HG03578'), 1),\n",
       " (('HG03582', 'HG03583'), 1),\n",
       " (('HG03594', 'HG03595'), 1),\n",
       " (('HG03606', 'HG03607'), 1),\n",
       " (('HG03615', 'HG03616'), 1),\n",
       " (('HG03618', 'HG03619'), 1),\n",
       " (('HG03624', 'HG03625'), 1),\n",
       " (('HG03633', 'HG03634'), 1),\n",
       " (('HG03639', 'HG03640'), 1),\n",
       " (('HG03649', 'HG03650'), 1),\n",
       " (('HG03652', 'HG03653'), 1),\n",
       " (('HG03667', 'HG03668'), 1),\n",
       " (('HG03682', 'HG03684'), 1),\n",
       " (('HG03885', 'HG03886'), 1),\n",
       " (('HG03699', 'HG03700'), 1),\n",
       " (('HG03702', 'HG03703'), 1),\n",
       " (('HG03705', 'HG03706'), 1),\n",
       " (('HG03708', 'HG03709'), 1),\n",
       " (('HG03725', 'HG03722'), 1),\n",
       " (('HG03727', 'HG03721'), 1),\n",
       " (('HG03761', 'HG03762'), 1),\n",
       " (('HG03794', 'HG03793'), 1),\n",
       " (('HG03797', 'HG03796'), 1),\n",
       " (('HG03800', 'HG03799'), 1),\n",
       " (('HG03803', 'HG03802'), 1),\n",
       " (('HG03806', 'HG03805'), 1),\n",
       " (('HG03815', 'HG03814'), 1),\n",
       " (('HG03824', 'HG03823'), 1),\n",
       " (('HG03830', 'HG03829'), 1),\n",
       " (('HG03833', 'HG03832'), 1),\n",
       " (('HG03905', 'HG03904'), 1),\n",
       " (('HG03908', 'HG03907'), 1),\n",
       " (('HG03914', 'HG03913'), 1),\n",
       " (('HG03926', 'HG03925'), 1),\n",
       " (('HG03929', 'HG03928'), 1),\n",
       " (('HG03941', 'HG03940'), 1),\n",
       " (('0', 'HG04067'), 1),\n",
       " (('HG03896', 'HG03894'), 1),\n",
       " (('HG04127', 'HG04075'), 1),\n",
       " (('HG04131', 'HG04132'), 1),\n",
       " (('HG04134', 'HG04135'), 1),\n",
       " (('HG04140', 'HG04141'), 1),\n",
       " (('HG04146', 'HG04147'), 1),\n",
       " (('HG04149', 'HG04150'), 1),\n",
       " (('HG04155', 'HG04156'), 1),\n",
       " (('HG04158', 'HG04159'), 1),\n",
       " (('HG04173', 'HG04174'), 1),\n",
       " (('HG04182', 'HG04183'), 1),\n",
       " (('HG04185', 'HG04186'), 1),\n",
       " (('HG04191', 'HG04192'), 1),\n",
       " (('HG04100', 'HG04099'), 1),\n",
       " (('HG04198', 'HG04200'), 1),\n",
       " (('HG04229', 'HG04227'), 1),\n",
       " (('NA06993', 'NA06985'), 1),\n",
       " (('NA07435', 'NA07037'), 1),\n",
       " (('NA06986', 'NA07045'), 1),\n",
       " (('NA07051', 'NA07031'), 1),\n",
       " (('NA07022', 'NA07056'), 1),\n",
       " (('NA06994', 'NA07000'), 1),\n",
       " (('NA07034', 'NA07055'), 1),\n",
       " (('NA07357', 'NA07345'), 1),\n",
       " (('NA07347', 'NA07346'), 1),\n",
       " (('NA12154', 'NA12236'), 1),\n",
       " (('NA12155', 'NA12156'), 1),\n",
       " (('NA12248', 'NA12249'), 1),\n",
       " (('NA12274', 'NA12275'), 1),\n",
       " (('NA12272', 'NA12273'), 1),\n",
       " (('NA12003', 'NA12004'), 1),\n",
       " (('NA12005', 'NA12006'), 1),\n",
       " (('NA12286', 'NA12287'), 1),\n",
       " (('NA11917', 'NA11918'), 1),\n",
       " (('NA11919', 'NA11920'), 1),\n",
       " (('NA11930', 'NA11931'), 1),\n",
       " (('NA12144', 'NA12145'), 1),\n",
       " (('NA12146', 'NA12239'), 1),\n",
       " (('NA12056', 'NA12057'), 1),\n",
       " (('NA12045', 'NA12046'), 1),\n",
       " (('NA11839', 'NA11840'), 1),\n",
       " (('NA11831', 'NA11832'), 1),\n",
       " (('NA11829', 'NA11830'), 1),\n",
       " (('NA12043', 'NA12044'), 1),\n",
       " (('NA11881', 'NA11882'), 1),\n",
       " (('NA11992', 'NA11993'), 1),\n",
       " (('NA11994', 'NA11995'), 1),\n",
       " (('NA12264', 'NA12234'), 1),\n",
       " (('NA11893', 'NA11894'), 1),\n",
       " (('NA11891', 'NA11892'), 1),\n",
       " (('NA06984', 'NA06989'), 1),\n",
       " (('NA12340', 'NA12341'), 1),\n",
       " (('NA12342', 'NA12343'), 1),\n",
       " (('NA12347', 'NA12348'), 1),\n",
       " (('NA12546', 'NA12489'), 1),\n",
       " (('NA12399', 'NA12400'), 1),\n",
       " (('NA12413', 'NA12414'), 1),\n",
       " (('NA12716', 'NA12717'), 1),\n",
       " (('NA12748', 'NA12749'), 1),\n",
       " (('NA12750', 'NA12751'), 1),\n",
       " (('NA12760', 'NA12761'), 1),\n",
       " (('NA12762', 'NA12763'), 1),\n",
       " (('NA12775', 'NA12776'), 1),\n",
       " (('NA12777', 'NA12778'), 1),\n",
       " (('NA12812', 'NA12813'), 1),\n",
       " (('NA12814', 'NA12815'), 1),\n",
       " (('NA12827', 'NA12828'), 1),\n",
       " (('NA12829', 'NA12830'), 1),\n",
       " (('NA12842', 'NA12843'), 1),\n",
       " (('NA12872', 'NA12873'), 1),\n",
       " (('NA12874', 'NA12875'), 1),\n",
       " (('NA12889', 'NA12890'), 1),\n",
       " (('NA12891', 'NA12892'), 1),\n",
       " (('NA18486', 'NA18488'), 1),\n",
       " (('NA18487', 'NA18489'), 1),\n",
       " (('NA18498', 'NA18499'), 1),\n",
       " (('NA18501', 'NA18502'), 1),\n",
       " (('NA18504', 'NA18505'), 1),\n",
       " (('NA18507', 'NA18508'), 1),\n",
       " (('NA18516', 'NA18517'), 1),\n",
       " (('NA18519', 'NA18520'), 1),\n",
       " (('NA18522', 'NA18523'), 1),\n",
       " (('NA18853', 'NA18852'), 1),\n",
       " (('NA18856', 'NA18855'), 1),\n",
       " (('NA18859', 'NA18858'), 1),\n",
       " (('0', 'NA19105'), 1),\n",
       " (('NA18862', 'NA18861'), 1),\n",
       " (('NA18868', 'NA18867'), 1),\n",
       " (('NA18871', 'NA18870'), 1),\n",
       " (('NA18874', 'NA18873'), 1),\n",
       " (('NA18877', 'NA18876'), 1),\n",
       " (('NA18910', 'NA18909'), 1),\n",
       " (('0', 'NA19238'), 1),\n",
       " (('NA18913', 'NA18912'), 1),\n",
       " (('NA18923', 'NA18924'), 1),\n",
       " (('NA18917', 'NA18916'), 1),\n",
       " (('NA18934', 'NA18933'), 1),\n",
       " (('NA19092', 'NA19093'), 1),\n",
       " (('NA19096', 'NA19095'), 1),\n",
       " (('NA19098', 'NA19099'), 1),\n",
       " (('NA19101', 'NA19102'), 1),\n",
       " (('NA19107', 'NA19108'), 1),\n",
       " (('NA19113', 'NA19114'), 1),\n",
       " (('NA19119', 'NA19116'), 1),\n",
       " (('NA19121', 'NA19122'), 1),\n",
       " (('NA19128', 'NA19127'), 1),\n",
       " (('NA19130', 'NA19131'), 1),\n",
       " (('NA19138', 'NA19137'), 1),\n",
       " (('NA19141', 'NA19140'), 1),\n",
       " (('NA19144', 'NA19143'), 1),\n",
       " (('NA19146', 'NA19147'), 1),\n",
       " (('NA19150', 'NA19149'), 1),\n",
       " (('NA19153', 'NA19152'), 1),\n",
       " (('NA19160', 'NA19159'), 1),\n",
       " (('NA19171', 'NA19172'), 1),\n",
       " (('NA19117', 'NA19118'), 1),\n",
       " (('NA19175', 'NA19176'), 1),\n",
       " (('NA19184', 'NA19185'), 1),\n",
       " (('NA19189', 'NA19190'), 1),\n",
       " (('NA19198', 'NA19197'), 1),\n",
       " (('NA19200', 'NA19201'), 1),\n",
       " (('NA19203', 'NA19204'), 1),\n",
       " (('NA19207', 'NA19206'), 1),\n",
       " (('NA19210', 'NA19209'), 1),\n",
       " (('NA19213', 'NA19214'), 1),\n",
       " (('NA19223', 'NA19222'), 1),\n",
       " (('NA19226', 'NA19225'), 1),\n",
       " (('NA19236', 'NA19235'), 1),\n",
       " (('NA19239', 'NA19238'), 1),\n",
       " (('NA19248', 'NA19247'), 1),\n",
       " (('NA19256', 'NA19257'), 1),\n",
       " (('0', 'NA19432'), 1),\n",
       " (('NA19453', '0'), 1),\n",
       " (('NA19649', 'NA19648'), 1),\n",
       " (('NA19652', 'NA19651'), 1),\n",
       " (('NA19655', 'NA19654'), 1),\n",
       " (('NA19658', 'NA19657'), 1),\n",
       " (('NA19664', 'NA19663'), 1),\n",
       " (('NA19670', 'NA19669'), 1),\n",
       " (('NA19676', 'NA19675'), 1),\n",
       " (('NA19682', 'NA19681'), 1),\n",
       " (('NA19685', 'NA19684'), 1),\n",
       " (('NA19700', 'NA19701'), 1),\n",
       " (('NA19703', 'NA19704'), 1),\n",
       " (('NA19717', 'NA19716'), 1),\n",
       " (('NA19720', 'NA19719'), 1),\n",
       " (('NA19723', 'NA19722'), 1),\n",
       " (('NA19726', 'NA19725'), 1),\n",
       " (('NA19729', 'NA19728'), 1),\n",
       " (('NA19732', 'NA19731'), 1),\n",
       " (('NA19747', 'NA19746'), 1),\n",
       " (('NA19750', 'NA19749'), 1),\n",
       " (('NA19756', 'NA19755'), 1),\n",
       " (('NA19759', 'NA19758'), 1),\n",
       " (('NA19762', 'NA19761'), 1),\n",
       " (('NA19771', 'NA19770'), 1),\n",
       " (('NA19774', 'NA19773'), 1),\n",
       " (('NA19777', 'NA19776'), 1),\n",
       " (('NA19780', 'NA19779'), 1),\n",
       " (('NA19783', 'NA19782'), 1),\n",
       " (('NA19786', 'NA19785'), 1),\n",
       " (('NA19789', 'NA19788'), 1),\n",
       " (('NA19795', 'NA19794'), 1),\n",
       " (('NA19818', 'NA19819'), 1),\n",
       " (('NA19834', 'NA19835'), 1),\n",
       " (('NA19900', 'NA19901'), 1),\n",
       " (('NA19916', 'NA19917'), 1),\n",
       " (('NA19908', 'NA19909'), 1),\n",
       " (('NA19922', 'NA19923'), 1),\n",
       " (('NA19982', 'NA19713'), 1),\n",
       " (('NA20126', 'NA20127'), 1),\n",
       " (('NA19920', 'NA19921'), 1),\n",
       " (('NA20278', 'NA20282'), 1),\n",
       " (('NA20356', 'NA20357'), 1),\n",
       " (('NA20909', '0'), 1)]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter([(i,j) for i,j in zip(kg_ped.FatherID.values, kg_ped.MotherID.values)]).most_common()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FamilyID</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>FatherID</th>\n",
       "      <th>MotherID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Population</th>\n",
       "      <th>Superpopulation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>GBR001</td>\n",
       "      <td>HG00155</td>\n",
       "      <td>0</td>\n",
       "      <td>HG00144</td>\n",
       "      <td>1</td>\n",
       "      <td>GBR</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FamilyID SampleID FatherID MotherID  Sex Population Superpopulation\n",
       "50   GBR001  HG00155        0  HG00144    1        GBR             EUR"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kg_ped[kg_ped.FamilyID=='GBR001']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To copy/paste to pileup bash script.\n",
    "kg_files = pd.read_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/1kg_hg38_ids.txt', sep='\\t', header=None)\n",
    "kg_metadata = pd.read_csv('/scratch/groups/dpwall/personal/briannac/unmapped_reads/data/1kg_hg38_metadata.tsv', sep='\\t')\n",
    "kg_files.index = kg_files[9]\n",
    "kg_metadata.index=kg_metadata['Sample name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HG00766</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>da0e327c506d7bf6f1cd7c7a52be1917</td>\n",
       "      <td>ERR3242129</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1783865</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS179212</td>\n",
       "      <td>HG00766</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG00766</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:39:40:886-471</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>747102928</td>\n",
       "      <td>112065439200</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA12400</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR323...</td>\n",
       "      <td>c09b439ab04568b8bf67656f055d1b86</td>\n",
       "      <td>ERR3239478</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1783081</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS000629</td>\n",
       "      <td>NA12400</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>NA12400</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:30:23:667-203</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>898974846</td>\n",
       "      <td>134846226900</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA19700</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR323...</td>\n",
       "      <td>26b9fdfd02c239bc043d4578bf52c98b</td>\n",
       "      <td>ERR3239913</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1783410</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS003668</td>\n",
       "      <td>NA19700</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>NA19700</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:33:38:349-139</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>744943548</td>\n",
       "      <td>111741532200</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG02178</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>d45cd3ebcd463dfc17e2e2e23899fd2e</td>\n",
       "      <td>ERR3242174</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1784271</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS179259</td>\n",
       "      <td>HG02178</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG02178</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:43:59:740-16</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>737648030</td>\n",
       "      <td>110647204500</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG00500</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>f5bfe1b900752ae38493c39a2ac1e3ad</td>\n",
       "      <td>ERR3241683</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1783865</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS008628</td>\n",
       "      <td>HG00500</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG00500</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:39:40:869-25</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>806376892</td>\n",
       "      <td>120956533800</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG00632</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>f754219db5c3159967b4e426a80ee7b3</td>\n",
       "      <td>ERR3242520</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1784271</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS344120</td>\n",
       "      <td>HG00632</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG00632</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:43:59:752-362</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>700382870</td>\n",
       "      <td>105057430500</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG03199</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>209a463dcb4473ea7e49bcf520e9f2f4</td>\n",
       "      <td>ERR3242621</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1784271</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS350784</td>\n",
       "      <td>HG03199</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG03199</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:43:59:755-463</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>738182354</td>\n",
       "      <td>110727353100</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG03727</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>5c773cf81989486018da264987e20901</td>\n",
       "      <td>ERR3242704</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1784521</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS350903</td>\n",
       "      <td>HG03727</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG03727</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:47:51:973-46</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>791610534</td>\n",
       "      <td>118741580100</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG02274</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>9637eaef4e36ac2444ce8775fa4b59db</td>\n",
       "      <td>ERR3242821</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1784521</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS351035</td>\n",
       "      <td>HG02274</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG02274</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:47:51:977-163</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>772875158</td>\n",
       "      <td>115931273700</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG01049</th>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...</td>\n",
       "      <td>1790d56d751786c5d9324f55eaf5cc95</td>\n",
       "      <td>ERR3241762</td>\n",
       "      <td>ERP114329</td>\n",
       "      <td>30X whole genome sequencing coverage of the 25...</td>\n",
       "      <td>NYGC</td>\n",
       "      <td>ERA1783865</td>\n",
       "      <td>2019-03-22 00:00</td>\n",
       "      <td>SRS008756</td>\n",
       "      <td>HG01049</td>\n",
       "      <td>...</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>HG01049</td>\n",
       "      <td>ena-RUN-NYGC-22-03-2019-12:39:40:872-104</td>\n",
       "      <td>450</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>894618654</td>\n",
       "      <td>134192798100</td>\n",
       "      <td>high_cov</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2501 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        0   \\\n",
       "9                                                            \n",
       "HG00766  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "NA12400  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR323...   \n",
       "NA19700  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR323...   \n",
       "HG02178  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "HG00500  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "...                                                    ...   \n",
       "HG00632  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "HG03199  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "HG03727  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "HG02274  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "HG01049  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR324...   \n",
       "\n",
       "                                       1           2          3   \\\n",
       "9                                                                  \n",
       "HG00766  da0e327c506d7bf6f1cd7c7a52be1917  ERR3242129  ERP114329   \n",
       "NA12400  c09b439ab04568b8bf67656f055d1b86  ERR3239478  ERP114329   \n",
       "NA19700  26b9fdfd02c239bc043d4578bf52c98b  ERR3239913  ERP114329   \n",
       "HG02178  d45cd3ebcd463dfc17e2e2e23899fd2e  ERR3242174  ERP114329   \n",
       "HG00500  f5bfe1b900752ae38493c39a2ac1e3ad  ERR3241683  ERP114329   \n",
       "...                                   ...         ...        ...   \n",
       "HG00632  f754219db5c3159967b4e426a80ee7b3  ERR3242520  ERP114329   \n",
       "HG03199  209a463dcb4473ea7e49bcf520e9f2f4  ERR3242621  ERP114329   \n",
       "HG03727  5c773cf81989486018da264987e20901  ERR3242704  ERP114329   \n",
       "HG02274  9637eaef4e36ac2444ce8775fa4b59db  ERR3242821  ERP114329   \n",
       "HG01049  1790d56d751786c5d9324f55eaf5cc95  ERR3241762  ERP114329   \n",
       "\n",
       "                                                        4     5           6   \\\n",
       "9                                                                              \n",
       "HG00766  30X whole genome sequencing coverage of the 25...  NYGC  ERA1783865   \n",
       "NA12400  30X whole genome sequencing coverage of the 25...  NYGC  ERA1783081   \n",
       "NA19700  30X whole genome sequencing coverage of the 25...  NYGC  ERA1783410   \n",
       "HG02178  30X whole genome sequencing coverage of the 25...  NYGC  ERA1784271   \n",
       "HG00500  30X whole genome sequencing coverage of the 25...  NYGC  ERA1783865   \n",
       "...                                                    ...   ...         ...   \n",
       "HG00632  30X whole genome sequencing coverage of the 25...  NYGC  ERA1784271   \n",
       "HG03199  30X whole genome sequencing coverage of the 25...  NYGC  ERA1784271   \n",
       "HG03727  30X whole genome sequencing coverage of the 25...  NYGC  ERA1784521   \n",
       "HG02274  30X whole genome sequencing coverage of the 25...  NYGC  ERA1784521   \n",
       "HG01049  30X whole genome sequencing coverage of the 25...  NYGC  ERA1783865   \n",
       "\n",
       "                       7          8        9   ...        12  \\\n",
       "9                                              ...             \n",
       "HG00766  2019-03-22 00:00  SRS179212  HG00766  ...  ILLUMINA   \n",
       "NA12400  2019-03-22 00:00  SRS000629  NA12400  ...  ILLUMINA   \n",
       "NA19700  2019-03-22 00:00  SRS003668  NA19700  ...  ILLUMINA   \n",
       "HG02178  2019-03-22 00:00  SRS179259  HG02178  ...  ILLUMINA   \n",
       "HG00500  2019-03-22 00:00  SRS008628  HG00500  ...  ILLUMINA   \n",
       "...                   ...        ...      ...  ...       ...   \n",
       "HG00632  2019-03-22 00:00  SRS344120  HG00632  ...  ILLUMINA   \n",
       "HG03199  2019-03-22 00:00  SRS350784  HG03199  ...  ILLUMINA   \n",
       "HG03727  2019-03-22 00:00  SRS350903  HG03727  ...  ILLUMINA   \n",
       "HG02274  2019-03-22 00:00  SRS351035  HG02274  ...  ILLUMINA   \n",
       "HG01049  2019-03-22 00:00  SRS008756  HG01049  ...  ILLUMINA   \n",
       "\n",
       "                            13       14  \\\n",
       "9                                         \n",
       "HG00766  Illumina NovaSeq 6000  HG00766   \n",
       "NA12400  Illumina NovaSeq 6000  NA12400   \n",
       "NA19700  Illumina NovaSeq 6000  NA19700   \n",
       "HG02178  Illumina NovaSeq 6000  HG02178   \n",
       "HG00500  Illumina NovaSeq 6000  HG00500   \n",
       "...                        ...      ...   \n",
       "HG00632  Illumina NovaSeq 6000  HG00632   \n",
       "HG03199  Illumina NovaSeq 6000  HG03199   \n",
       "HG03727  Illumina NovaSeq 6000  HG03727   \n",
       "HG02274  Illumina NovaSeq 6000  HG02274   \n",
       "HG01049  Illumina NovaSeq 6000  HG01049   \n",
       "\n",
       "                                               15   16      17  18         19  \\\n",
       "9                                                                               \n",
       "HG00766  ena-RUN-NYGC-22-03-2019-12:39:40:886-471  450  PAIRED NaN  747102928   \n",
       "NA12400  ena-RUN-NYGC-22-03-2019-12:30:23:667-203  450  PAIRED NaN  898974846   \n",
       "NA19700  ena-RUN-NYGC-22-03-2019-12:33:38:349-139  450  PAIRED NaN  744943548   \n",
       "HG02178   ena-RUN-NYGC-22-03-2019-12:43:59:740-16  450  PAIRED NaN  737648030   \n",
       "HG00500   ena-RUN-NYGC-22-03-2019-12:39:40:869-25  450  PAIRED NaN  806376892   \n",
       "...                                           ...  ...     ...  ..        ...   \n",
       "HG00632  ena-RUN-NYGC-22-03-2019-12:43:59:752-362  450  PAIRED NaN  700382870   \n",
       "HG03199  ena-RUN-NYGC-22-03-2019-12:43:59:755-463  450  PAIRED NaN  738182354   \n",
       "HG03727   ena-RUN-NYGC-22-03-2019-12:47:51:973-46  450  PAIRED NaN  791610534   \n",
       "HG02274  ena-RUN-NYGC-22-03-2019-12:47:51:977-163  450  PAIRED NaN  772875158   \n",
       "HG01049  ena-RUN-NYGC-22-03-2019-12:39:40:872-104  450  PAIRED NaN  894618654   \n",
       "\n",
       "                   20        21  \n",
       "9                                \n",
       "HG00766  112065439200  high_cov  \n",
       "NA12400  134846226900  high_cov  \n",
       "NA19700  111741532200  high_cov  \n",
       "HG02178  110647204500  high_cov  \n",
       "HG00500  120956533800  high_cov  \n",
       "...               ...       ...  \n",
       "HG00632  105057430500  high_cov  \n",
       "HG03199  110727353100  high_cov  \n",
       "HG03727  118741580100  high_cov  \n",
       "HG02274  115931273700  high_cov  \n",
       "HG01049  134192798100  high_cov  \n",
       "\n",
       "[2501 rows x 22 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids = list(set(kg_files.index).intersection(set(kg_metadata.index)))\n",
    "kg_files = kg_files.loc[ids]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242773/HG01092.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242646/HG03793.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242946/HG02879.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242246/HG02292.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239511/NA18613.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3241952/HG01617.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242657/HG03829.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239713/NA19318.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242102/HG02049.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3243044/HG03767.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239774/NA19461.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3240167/HG00180.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239592/NA19060.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239705/NA19308.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239417/NA18975.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242617/HG03168.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3242593/HG03291.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR324/ERR3240095/NA19922.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239835/NA20755.final.cram' 'ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR323/ERR3239672/NA18950.final.cram\""
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids = list(set(kg_files.index).intersection(set(kg_metadata.index)))\n",
    "kg_files = kg_files.loc[ids]\n",
    "kg_metadata = kg_metadata.loc[ids]\n",
    "\"' '\".join([i for i in kg_files[0].values[:20]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'HG01092 HG03793 HG02879 HG02292 NA18613 HG01617 HG03829 NA19318 HG02049 HG03767 NA19461 HG00180 NA19060 NA19308 NA18975 HG03168 HG03291 NA19922 NA20755 NA18950'"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "' '.join([i for i in kg_files[9].values[:20]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'female': 11, 'male': 9})"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(kg_metadata.Sex[:20].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'2.0': 30, '1.0': 9})"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(bam_mappings.loc[[\"07C65108\",\"09C82723\",\"07C70153\",\"10C106894\",\"10C106929\",\"07C68567\",\"07C63191\",\"08C76306\",\"10C115947\",\"08C76307\",\"07C65814\",\"10C106936\",\"10C117064\",\"07C70151\",\"09C87488\",\"10C115948\",\"10C117063\",\"07C63192\",\"07C70366\",\"09C82721\",\"07C65105\",\"10C108230\",\"07C70372\",\"07C68568\",\"07C65107\",\"07C63425\",\"09C82722\",\"09C79835\",\"10C106892\",\"07C68566\",\"07C65104\",\"10C108279\",\"10C115946\",\"07C70365\",\"10C106933\",\"07C65103\",\"07C65818\",\"09C82724\",\"10C106893\"]\n",
    "                ].sex_numeric.values)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Batch 1\n",
    "sequencing_place = MTR-00027"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'07C65108\",\"09C82723\",\"07C70153\",\"10C106894\",\"10C106929\",\"07C68567\",\"07C63191\",\"08C76306\",\"10C115947\",\"08C76307\",\"07C65814\",\"10C106936\",\"10C117064\",\"07C70151\",\"09C87488\",\"10C115948\",\"10C117063\",\"07C63192\",\"07C70366\",\"09C82721\",\"07C65105\",\"10C108230\",\"07C70372\",\"07C68568\",\"07C65107\",\"07C63425\",\"09C82722\",\"09C79835\",\"10C106892\",\"07C68566\",\"07C65104\",\"10C108279\",\"10C115946\",\"07C70365\",\"10C106933\",\"07C65103\",\"07C65818\",\"09C82724\",\"10C106893'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'07C65108 09C82723 07C70153 10C106894 10C106929 07C68567 07C63191 08C76306 10C115947 08C76307 07C65814 10C106936 10C117064 07C70151 09C87488 10C115948 10C117063 07C63192 07C70366 09C82721 07C65105 10C108230 07C70372 07C68568 07C65107 07C63425 09C82722 09C79835 10C106892 07C68566 07C65104 10C108279 10C115946 07C70365 10C106933 07C65103 07C65818 09C82724 10C106893'.replace(' ', '\",\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "95  people in batch\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'03C14328 03C15416 03C15417 03C15418 03C15419 03C15790 03C15791 03C15962 03C15963 03C16097 03C16098 03C16794 03C16795 03C16796 03C16797 03C16798 10C102514 10C102515 10C102516 10C102517 10C103890 10C103891 10C103892 10C103893 10C104027 10C104028 10C104029 10C104030 10C104031 10C105023 10C110447 10C110448 10C110449 10C110450 10C110451 10C110758 10C110759 10C110760 10C110761 10C112856 10C112857 10C112858 10C112859 10C112904 11C120670 11C120671 11C120680 11C120681 11C120684 11C120729 11C120730 11C120731 11C120732 11C122939 11C122940 11C122941 11C122942 11C123571 11C123572 11C123573 11C123574 11C123749 11C123777 11C123778 11C125629 11C125630 11C125631 11C125632 11C125633 11C125634 11C125860 11C125861 11C125862 11C125875 11C125876 MH0131344 MH0131348 MH0131365 MH0135248 MH0137425 MH0137426 MH0137427 MH0137428 MH0137429 MH0138049 MH0138050 MH0138051 MH0138052 MH0138054 MH0138055 MH0143008 MH0143009 MH0143013 MH0143018 MH0143019'"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(bam_mappings[bam_mappings.sequencing_plate=='MTR-00027'].sample_id.values), ' people in batch')\n",
    "' '.join(bam_mappings[bam_mappings.sequencing_plate=='MTR-00027'].sample_id.values)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Family 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>status</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>family</th>\n",
       "      <th>AGRE Blood</th>\n",
       "      <th>NIMH Blood</th>\n",
       "      <th>relationship</th>\n",
       "      <th>FatherID</th>\n",
       "      <th>MotherID</th>\n",
       "      <th>sex_numeric</th>\n",
       "      <th>...</th>\n",
       "      <th>number_of_indels</th>\n",
       "      <th>number_of_homozygous_snvs</th>\n",
       "      <th>number_of_heterozygous_snvs</th>\n",
       "      <th>het_to_hom_ratio</th>\n",
       "      <th>ti_tv_ratio</th>\n",
       "      <th>dbsnp_overlap</th>\n",
       "      <th>concordance_with_genotyping_chip</th>\n",
       "      <th>cryptic_relatedness_check</th>\n",
       "      <th>family_based_average_coverage</th>\n",
       "      <th>familial_relationship_check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4837</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>MH0143008</td>\n",
       "      <td>AU3859201</td>\n",
       "      <td>AU3859</td>\n",
       "      <td>HI11780</td>\n",
       "      <td>MH0143008</td>\n",
       "      <td>father</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4838</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>MH0143009</td>\n",
       "      <td>AU3859202</td>\n",
       "      <td>AU3859</td>\n",
       "      <td>HI11781</td>\n",
       "      <td>MH0143009</td>\n",
       "      <td>mother</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4839</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>MH0143013</td>\n",
       "      <td>AU3859303</td>\n",
       "      <td>AU3859</td>\n",
       "      <td>HI11784</td>\n",
       "      <td>MH0143013</td>\n",
       "      <td>sibling</td>\n",
       "      <td>AU3859201</td>\n",
       "      <td>AU3859202</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4840</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>MH0143018</td>\n",
       "      <td>AU3859301</td>\n",
       "      <td>AU3859</td>\n",
       "      <td>HI11782</td>\n",
       "      <td>MH0143018</td>\n",
       "      <td>sibling</td>\n",
       "      <td>AU3859201</td>\n",
       "      <td>AU3859202</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4841</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>MH0143019</td>\n",
       "      <td>AU3859302</td>\n",
       "      <td>AU3859</td>\n",
       "      <td>HI11783</td>\n",
       "      <td>MH0143019</td>\n",
       "      <td>sibling</td>\n",
       "      <td>AU3859201</td>\n",
       "      <td>AU3859202</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        status  sample_id participant_id  family AGRE Blood  \\\n",
       "4837  Passed_QC_analysis_ready  MH0143008      AU3859201  AU3859    HI11780   \n",
       "4838  Passed_QC_analysis_ready  MH0143009      AU3859202  AU3859    HI11781   \n",
       "4839  Passed_QC_analysis_ready  MH0143013      AU3859303  AU3859    HI11784   \n",
       "4840  Passed_QC_analysis_ready  MH0143018      AU3859301  AU3859    HI11782   \n",
       "4841  Passed_QC_analysis_ready  MH0143019      AU3859302  AU3859    HI11783   \n",
       "\n",
       "     NIMH Blood relationship   FatherID   MotherID sex_numeric  ...  \\\n",
       "4837  MH0143008       father          0          0         1.0  ...   \n",
       "4838  MH0143009       mother          0          0         2.0  ...   \n",
       "4839  MH0143013      sibling  AU3859201  AU3859202         1.0  ...   \n",
       "4840  MH0143018      sibling  AU3859201  AU3859202         1.0  ...   \n",
       "4841  MH0143019      sibling  AU3859201  AU3859202         2.0  ...   \n",
       "\n",
       "     number_of_indels number_of_homozygous_snvs  number_of_heterozygous_snvs  \\\n",
       "4837              NaN                       NaN                          NaN   \n",
       "4838              NaN                       NaN                          NaN   \n",
       "4839              NaN                       NaN                          NaN   \n",
       "4840              NaN                       NaN                          NaN   \n",
       "4841              NaN                       NaN                          NaN   \n",
       "\n",
       "     het_to_hom_ratio ti_tv_ratio dbsnp_overlap  \\\n",
       "4837              NaN         NaN           NaN   \n",
       "4838              NaN         NaN           NaN   \n",
       "4839              NaN         NaN           NaN   \n",
       "4840              NaN         NaN           NaN   \n",
       "4841              NaN         NaN           NaN   \n",
       "\n",
       "     concordance_with_genotyping_chip cryptic_relatedness_check  \\\n",
       "4837                              NaN                       NaN   \n",
       "4838                              NaN                       NaN   \n",
       "4839                              NaN                       NaN   \n",
       "4840                              NaN                       NaN   \n",
       "4841                              NaN                       NaN   \n",
       "\n",
       "     family_based_average_coverage familial_relationship_check  \n",
       "4837                           NaN                         NaN  \n",
       "4838                           NaN                         NaN  \n",
       "4839                           NaN                         NaN  \n",
       "4840                           NaN                         NaN  \n",
       "4841                           NaN                         NaN  \n",
       "\n",
       "[5 rows x 74 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam = bam_mappings[bam_mappings['sample_id']=='MH0143019']['family'].values[0]\n",
    "bam_mappings[bam_mappings['family']==fam]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "family 1: MH0143008 MH0143009 MH0143013 MH0143018 MH0143019\n",
    "\n",
    "**Note: These are WB**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Family 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>status</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>family</th>\n",
       "      <th>AGRE Blood</th>\n",
       "      <th>NIMH Blood</th>\n",
       "      <th>relationship</th>\n",
       "      <th>FatherID</th>\n",
       "      <th>MotherID</th>\n",
       "      <th>sex_numeric</th>\n",
       "      <th>...</th>\n",
       "      <th>number_of_indels</th>\n",
       "      <th>number_of_homozygous_snvs</th>\n",
       "      <th>number_of_heterozygous_snvs</th>\n",
       "      <th>het_to_hom_ratio</th>\n",
       "      <th>ti_tv_ratio</th>\n",
       "      <th>dbsnp_overlap</th>\n",
       "      <th>concordance_with_genotyping_chip</th>\n",
       "      <th>cryptic_relatedness_check</th>\n",
       "      <th>family_based_average_coverage</th>\n",
       "      <th>familial_relationship_check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>02C10540</td>\n",
       "      <td>AU1005201</td>\n",
       "      <td>AU1005</td>\n",
       "      <td>HI2393</td>\n",
       "      <td>02C10540</td>\n",
       "      <td>father</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>02C10541</td>\n",
       "      <td>AU1005202</td>\n",
       "      <td>AU1005</td>\n",
       "      <td>HI2394</td>\n",
       "      <td>02C10541</td>\n",
       "      <td>mother</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>02C10542</td>\n",
       "      <td>AU1005302</td>\n",
       "      <td>AU1005</td>\n",
       "      <td>HI2395</td>\n",
       "      <td>02C10542</td>\n",
       "      <td>sibling</td>\n",
       "      <td>AU1005201</td>\n",
       "      <td>AU1005202</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>02C10543</td>\n",
       "      <td>AU1005301</td>\n",
       "      <td>AU1005</td>\n",
       "      <td>HI2396</td>\n",
       "      <td>02C10543</td>\n",
       "      <td>sibling</td>\n",
       "      <td>AU1005201</td>\n",
       "      <td>AU1005202</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     status sample_id participant_id  family AGRE Blood  \\\n",
       "0  Passed_QC_analysis_ready  02C10540      AU1005201  AU1005     HI2393   \n",
       "1  Passed_QC_analysis_ready  02C10541      AU1005202  AU1005     HI2394   \n",
       "2  Passed_QC_analysis_ready  02C10542      AU1005302  AU1005     HI2395   \n",
       "3  Passed_QC_analysis_ready  02C10543      AU1005301  AU1005     HI2396   \n",
       "\n",
       "  NIMH Blood relationship   FatherID   MotherID sex_numeric  ...  \\\n",
       "0   02C10540       father          0          0         1.0  ...   \n",
       "1   02C10541       mother          0          0         2.0  ...   \n",
       "2   02C10542      sibling  AU1005201  AU1005202         2.0  ...   \n",
       "3   02C10543      sibling  AU1005201  AU1005202         1.0  ...   \n",
       "\n",
       "  number_of_indels number_of_homozygous_snvs  number_of_heterozygous_snvs  \\\n",
       "0              NaN                       NaN                          NaN   \n",
       "1              NaN                       NaN                          NaN   \n",
       "2              NaN                       NaN                          NaN   \n",
       "3              NaN                       NaN                          NaN   \n",
       "\n",
       "  het_to_hom_ratio ti_tv_ratio dbsnp_overlap concordance_with_genotyping_chip  \\\n",
       "0              NaN         NaN           NaN                              NaN   \n",
       "1              NaN         NaN           NaN                              NaN   \n",
       "2              NaN         NaN           NaN                              NaN   \n",
       "3              NaN         NaN           NaN                              NaN   \n",
       "\n",
       "  cryptic_relatedness_check family_based_average_coverage  \\\n",
       "0                       NaN                           NaN   \n",
       "1                       NaN                           NaN   \n",
       "2                       NaN                           NaN   \n",
       "3                       NaN                           NaN   \n",
       "\n",
       "  familial_relationship_check  \n",
       "0                         NaN  \n",
       "1                         NaN  \n",
       "2                         NaN  \n",
       "3                         NaN  \n",
       "\n",
       "[4 rows x 74 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_mappings[bam_mappings['family']=='AU1005']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "family 2: 02C10540 02C10541 02C10542 02C10543\n",
    "\n",
    "**Note: These are LCL**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Family 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>family</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>relationship</th>\n",
       "      <th>sex_numeric</th>\n",
       "      <th>status</th>\n",
       "      <th>batch_name</th>\n",
       "      <th>sequencing_plate</th>\n",
       "      <th>bio_seq_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>AU0162</td>\n",
       "      <td>AU016204</td>\n",
       "      <td>03C14328</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>AU0162</td>\n",
       "      <td>AU016201</td>\n",
       "      <td>03C15416</td>\n",
       "      <td>mother</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>AU0162</td>\n",
       "      <td>AU016202</td>\n",
       "      <td>03C15417</td>\n",
       "      <td>father</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>AU0162</td>\n",
       "      <td>AU016203</td>\n",
       "      <td>03C15418</td>\n",
       "      <td>sibling</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>AU0162</td>\n",
       "      <td>AU016205</td>\n",
       "      <td>03C15419</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Sequencing_canceled</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>AU0236</td>\n",
       "      <td>AU023601</td>\n",
       "      <td>03C15790</td>\n",
       "      <td>mother</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>AU0236</td>\n",
       "      <td>AU023602</td>\n",
       "      <td>03C15791</td>\n",
       "      <td>father</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>AU0080</td>\n",
       "      <td>AU008001</td>\n",
       "      <td>03C15962</td>\n",
       "      <td>mother</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>AU0080</td>\n",
       "      <td>AU008002</td>\n",
       "      <td>03C15963</td>\n",
       "      <td>father</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>AU0080</td>\n",
       "      <td>AU008003</td>\n",
       "      <td>03C16097</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>AU0080</td>\n",
       "      <td>AU008004</td>\n",
       "      <td>03C16098</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>AU0041</td>\n",
       "      <td>AU004101</td>\n",
       "      <td>03C16794</td>\n",
       "      <td>mother</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654</th>\n",
       "      <td>AU0041</td>\n",
       "      <td>AU004102</td>\n",
       "      <td>03C16795</td>\n",
       "      <td>father</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>655</th>\n",
       "      <td>AU0041</td>\n",
       "      <td>AU004103</td>\n",
       "      <td>03C16796</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>656</th>\n",
       "      <td>AU0041</td>\n",
       "      <td>AU004104</td>\n",
       "      <td>03C16797</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>657</th>\n",
       "      <td>AU0041</td>\n",
       "      <td>AU004105</td>\n",
       "      <td>03C16798</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>LCL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4227</th>\n",
       "      <td>AU3154</td>\n",
       "      <td>AU3154201</td>\n",
       "      <td>10C102514</td>\n",
       "      <td>father</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>WB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4228</th>\n",
       "      <td>AU3154</td>\n",
       "      <td>AU3154202</td>\n",
       "      <td>10C102515</td>\n",
       "      <td>mother</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>WB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4229</th>\n",
       "      <td>AU3154</td>\n",
       "      <td>AU3154301</td>\n",
       "      <td>10C102516</td>\n",
       "      <td>sibling</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>WB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4231</th>\n",
       "      <td>AU3154</td>\n",
       "      <td>AU3154302</td>\n",
       "      <td>10C102517</td>\n",
       "      <td>sibling</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>WB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4257</th>\n",
       "      <td>AU3716</td>\n",
       "      <td>AU3716201</td>\n",
       "      <td>10C103890</td>\n",
       "      <td>father</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Passed_QC_analysis_ready</td>\n",
       "      <td>WAL10680_B01</td>\n",
       "      <td>MTR-00027</td>\n",
       "      <td>WB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      family participant_id  sample_id relationship sex_numeric  \\\n",
       "169   AU0162       AU016204   03C14328      sibling         2.0   \n",
       "230   AU0162       AU016201   03C15416       mother         2.0   \n",
       "231   AU0162       AU016202   03C15417       father         1.0   \n",
       "232   AU0162       AU016203   03C15418      sibling         1.0   \n",
       "233   AU0162       AU016205   03C15419      sibling         2.0   \n",
       "263   AU0236       AU023601   03C15790       mother         2.0   \n",
       "264   AU0236       AU023602   03C15791       father         1.0   \n",
       "302   AU0080       AU008001   03C15962       mother         2.0   \n",
       "303   AU0080       AU008002   03C15963       father         1.0   \n",
       "411   AU0080       AU008003   03C16097      sibling         2.0   \n",
       "412   AU0080       AU008004   03C16098      sibling         2.0   \n",
       "653   AU0041       AU004101   03C16794       mother         2.0   \n",
       "654   AU0041       AU004102   03C16795       father         1.0   \n",
       "655   AU0041       AU004103   03C16796      sibling         2.0   \n",
       "656   AU0041       AU004104   03C16797      sibling         2.0   \n",
       "657   AU0041       AU004105   03C16798      sibling         2.0   \n",
       "4227  AU3154      AU3154201  10C102514       father         1.0   \n",
       "4228  AU3154      AU3154202  10C102515       mother         2.0   \n",
       "4229  AU3154      AU3154301  10C102516      sibling         2.0   \n",
       "4231  AU3154      AU3154302  10C102517      sibling         1.0   \n",
       "4257  AU3716      AU3716201  10C103890       father         1.0   \n",
       "\n",
       "                        status    batch_name sequencing_plate bio_seq_source  \n",
       "169   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "230   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "231   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "232   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "233        Sequencing_canceled  WAL10680_B01        MTR-00027            LCL  \n",
       "263   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "264   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "302   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "303   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "411   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "412   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "653   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "654   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "655   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "656   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "657   Passed_QC_analysis_ready  WAL10680_B01        MTR-00027            LCL  \n",
       "4227  Passed_QC_analysis_ready  WAL10680_B01        MTR-00027             WB  \n",
       "4228  Passed_QC_analysis_ready  WAL10680_B01        MTR-00027             WB  \n",
       "4229  Passed_QC_analysis_ready  WAL10680_B01        MTR-00027             WB  \n",
       "4231  Passed_QC_analysis_ready  WAL10680_B01        MTR-00027             WB  \n",
       "4257  Passed_QC_analysis_ready  WAL10680_B01        MTR-00027             WB  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_plate =bam_mappings[bam_mappings['family']=='AU3859']['sequencing_plate'].values[0]\n",
    "bam_mappings[bam_mappings['sequencing_plate']==seq_plate].head(21)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using family AU0041, another family from AU3859's/MH0143019's batch  ('MTR-00027')\n",
    "\n",
    "sample IDS: 03C16794, 03C16795, 03C16796, 03C16797, 03C16798\n",
    "\n",
    "**Note: These are LCL.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
